Source/x11/XGServerEvent.m

```diff
@@ -1912,27 +1912,42 @@ - (NSEvent *)_handleTakeFocusAtom: (XEvent)xEvent
                        forContext: (NSGraphicsContext *)gcontext
 {
   int key_num;
-  NSWindow *key_win;
+  NSWindow *keyWindow;
   NSEvent *e = nil;
-  key_win = [NSApp keyWindow];
-  key_num = [key_win windowNumber];
-  NSDebugLLog(@"Focus", @"take focus:%lu (current=%lu key=%d)",
-              cWin->number, generic.currentFocusWindow, key_num);
-
-  /* Sometimes window managers lose the setinputfocus on the key window
-   * e.g. when ordering out a window with focus then ordering in the key window.   
-   * it might search for a window until one accepts its take focus request.
-   */
-  if (key_num == cWin->number)
-    cWin->ignore_take_focus = NO;
+  gswindow_device_t *this_window;
+  gswindow_device_t *key_window;
   
+  keyWindow = [NSApp keyWindow];
+  key_num = [keyWindow windowNumber];
+  key_window = [XGServer _windowWithTag:key_num];
+  this_window = [XGServer _windowForXWindow:xEvent.xfocus.window];
```

Comment:
Using cwin instead of this_window is better as this is is already coached for all ClientMessage events.
In the following log message the tow should always produce the same number. I would suggest to stick with cwin as this reduces the size of the change and keeps the code closer to the surrounding.

Reply:
Thanks, done.